Artesian/_Services/Exceptions.py

```diff
@@ -0,0 +1,125 @@
+
+
+class ArtesianSdkException(Exception):
+    """
+        Base class for all `Artesian` errors.
+    """
+
+    def __init__(self, message: str) -> None:
+        """
+            Constructor.
+
+            Args:
+                message: the message describing the problem.
+        """
+        Exception.__init__(self, message)
+        self._message = message
+
+    @property
+    def message(self):
+        return self._message
+
+        
+class ArtesianSdkRemoteException(ArtesianSdkException):
+    """
+        Raised when the Artesian Service validation of an object failed.
+    """
+    def __init__(self, method: str, url: str, statusCode: int,  problemDetails: dict = None, errorText:str = None) -> None:
+        """
+            Constructor.
+
+            Args:
+                method: the HTTP method used
+                url: the url of the request
+                statusCode: the response status code
+                problemDetails: the returned problemDetails object (if any)
+                errorText: the response as text if problem details is not provided
+        """
+        self._method = method
+        self._url = url
+        self._statusCode = statusCode
+        self._problemDetails = problemDetails
+        self._errorText = errorText
+
+        if problemDetails is not None:
+            details = problemDetails.get('details', None) or problemDetails.get('title', None) or problemDetails.get('type', None)
```

Comment:
verify 'details' vs 'detail'

Reply:
rfc states this is 'detail' not 'details' 

please change also in the tests 

![Screenshot_20220202-164438](https://user-images.githubusercontent.com/5227688/152187277-5a091ca1-e264-43d1-8ce9-50a7f6b2d0ed.jpg)